File: resuelve-service/resuelve-service/backend/src/test/java/com/resuelve/unit/marca/ObtenerContenidoBienvenidaUCTest.java
package com.resuelve.unit.marca;

import com.resuelve.aplicacion.marca.ObtenerContenidoBienvenidaUC;
import com.resuelve.dominio.marca.ContenidoBienvenida;
import com.resuelve.dominio.marca.IContenidoBienvenidaRepositorio;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

@ExtendWith(MockitoExtension.class)
@DisplayName("ObtenerContenidoBienvenidaUC — pruebas unitarias")
class ObtenerContenidoBienvenidaUCTest {

    @Mock
    private IContenidoBienvenidaRepositorio repositorio;

    private ObtenerContenidoBienvenidaUC uc;

    @BeforeEach
    void setUp() {
        uc = new ObtenerContenidoBienvenidaUC(repositorio);
    }

    @Test
    @DisplayName("Dado que el repositorio retorna contenido activo, "
            + "Cuando se ejecuta el UC, "
            + "Entonces devuelve la entidad de dominio correctamente mapeada")
    void debeRetornarContenidoCuandoExisteRegistroActivo() {
        var contenidoEsperado = new ContenidoBienvenida(
                "https://assets.resuelve.com/logo.png",
                "https://assets.resuelve.com/video.mp4",
                "2.1.0"
        );
        when(repositorio.obtenerActivo()).thenReturn(Optional.of(contenidoEsperado));

        var resultado = uc.ejecutar();

        assertThat(resultado.logoUrl()).isEqualTo("https://assets.resuelve.com/logo.png");
        assertThat(resultado.videoUrl()).isEqualTo("https://assets.resuelve.com/video.mp4");
        assertThat(resultado.version()).isEqualTo("2.1.0");
        verify(repositorio).obtenerActivo();
    }

    @Test
    @DisplayName("Dado que el repositorio no retorna contenido, "
            + "Cuando se ejecuta el UC, "
            + "Entonces lanza ContenidoNoEncontradoException")
    void debeLanzarExcepcionCuandoNoExisteContenido() {
        when(repositorio.obtenerActivo()).thenReturn(Optional.empty());

        assertThatThrownBy(() -> uc.ejecutar())
                .isInstanceOf(com.resuelve.dominio.marca.ContenidoNoEncontradoException.class);
        verify(repositorio).obtenerActivo();
    }
}

File: resuelve-service/resuelve-service/mobile/test/unit/aplicacion/obtener_contenido_bienvenida_uc_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:mockito/annotations.dart';
import 'package:mockito/mockito.dart';
import 'package:resuelve_app/dominio/entidades/contenido_bienvenida.dart';
import 'package:resuelve_app/dominio/entidades/result.dart';
import 'package:resuelve_app/dominio/repositorios/i_contenido_marca_repositorio.dart';
import 'package:resuelve_app/aplicacion/casos_de_uso/obtener_contenido_bienvenida_uc.dart';

import 'obtener_contenido_bienvenida_uc_test.mocks.dart';

@GenerateMocks([IContenidoMarcaRepositorio])
void main() {
  late MockIContenidoMarcaRepositorio mockRepositorio;
  late ObtenerContenidoBienvenidaUC uc;

  setUp(() {
    mockRepositorio = MockIContenidoMarcaRepositorio();
    uc = ObtenerContenidoBienvenidaUC(repositorio: mockRepositorio);
  });

  group('ObtenerContenidoBienvenidaUC', () {
    test(
        'Dado que el repositorio retorna Success, '
        'Cuando se ejecuta el UC, '
        'Entonces delega al repositorio y retorna Result<ContenidoBienvenida>', () async {
      const contenido = ContenidoBienvenida(
        logoUrl: 'https://assets.resuelve.com/logo.png',
        videoUrl: 'https://assets.resuelve.com/video.mp4',
        version: '2.1.0',
      );
      when(mockRepositorio.obtener())
          .thenAnswer((_) async => const Success(contenido));

      final resultado = await uc.ejecutar();

      expect(resultado, isA<Success<ContenidoBienvenida>>());
      final success = resultado as Success<ContenidoBienvenida>;
      expect(success.value.logoUrl, 'https://assets.resuelve.com/logo.png');
      verify(mockRepositorio.obtener()).called(1);
    });

    test(
        'Dado que el repositorio retorna Failure, '
        'Cuando se ejecuta el UC, '
        'Entonces retorna Failure sin propagar excepción', () async {
      when(mockRepositorio.obtener())
          .thenAnswer((_) async => Failure(Exception('red no disponible')));

      final resultado = await uc.ejecutar();

      expect(resultado, isA<Failure<ContenidoBienvenida>>());
      verify(mockRepositorio.obtener()).called(1);
    });
  });
}
